docs/docs/tutorial/chapter2/side-quest.md

```diff
@@ -89,6 +133,55 @@ export const deletePost = ({ id }) => {
 }
 ```
 
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```javascript title="api/src/services/posts/posts.ts"
+import type { Prisma } from '@prisma/client'
+
+import { db } from 'src/lib/db'
+
+export const posts = () => {
+  return db.post.findMany()
+}
+
+export const post = ({ id }: Prisma.PostWhereUniqueInput) => {
+  return db.post.findUnique({
+    where: { id },
+  })
+}
+
+interface CreatePostArgs {
+  input: Prisma.PostCreateInput
+}
+
+export const createPost = ({ input }: CreatePostArgs) => {
+  return db.post.create({
+    data: input,
+  })
+}
+
+interface UpdatePostArgs extends Prisma.PostWhereUniqueInput {
+  input: Prisma.PostUpdateInput
+}
+
+export const updatePost = ({ id, input }: UpdatePostArgs) => {
+  return db.post.update({
+    data: input,
+    where: { id },
+  })
+}
+
+export const deletePost = ({ id }: Prisma.PostWhereUniqueInput) => {
+  return db.post.delete({
+    where: { id },
+  })
+}
+```
+
+</TabItem>
+</Tabs>
+
 > Helix/Envelop assumes these functions return promises, which `db` (an instance of `PrismaClient`) does. Helix/Envelop waits for them to resolve before responding with your query results, so you don't need to worry about `async`/`await` or mess with callbacks yourself.
```

Comment:
We should probably change these to `Yoga/Envelop` now?

Reply:
Done